plugins/modules/monitoring/datadog/datadog_downtime.py

```diff
@@ -0,0 +1,311 @@
+    id:
+        description:
+          - The identifier of the downtime.
+          - If empty, a new downtime gets created, otherwise it is either updated or deleted depending of the C(state).
+          - To keep your playbook idempotent, you should save the identifier in a file and read it in a lookup.
+        type: int
+    monitor_tags:
+        description:
+          - A list of monitor tags to which the downtime applies.
+          - The resulting downtime applies to monitors that match ALL provided monitor tags.
+        type: list
+        elements: str
+    scope:
+        description:
+          - A list of scopes to which the downtime applies.
+          - The resulting downtime applies to sources that matches ALL provided scopes.
+        type: list
+        elements: str
+    monitor_id:
+        description:
+          - The ID of the monitor to mute. If not provided, the downtime applies to all monitors.
+        type: int
+    downtime_message:
+        description:
+          - A message to include with notifications for this downtime.
+          - Email notifications can be sent to specific users by using the same "@username" notation as events.
+          - Monitor message template variables can be accessed by using double square brackets, i.e "[[" and "]]".
+        type: str
+    start:
+        type: int
+        description:
+          - POSIX timestamp to start the downtime. If not provided, the downtime starts the moment it is created.
+    end:
+        type: int
+        description:
+          - POSIX timestamp to end the downtime. If not provided, the downtime is in effect until you cancel it.
+    timezone:
+        description:
+          - The timezone for the downtime.
+        type: str
+    rrule:
+        description:
+          - The C(RRULE) standard for defining recurring events.
+          - For example, to have a recurring event on the first day of each month,
+          - select a type of rrule and set the C(FREQ) to C(MONTHLY) and C(BYMONTHDAY) to C(1).
+          - Most common rrule options from the iCalendar Spec are supported.
+          - Attributes specifying the duration in C(RRULE) are not supported (e.g. C(DTSTART), C(DTEND), C(DURATION)).
+        type: str
+"""
+
+EXAMPLES = """
+  - name: Create a downtime
+    register: downtime_var
+    community.general.datadog_downtime:
+      state: present
+      monitor_tags: ["foo:bar"]
+      downtime_message: "Downtime for foo:bar"
+      scope: "test"
+      api_key: "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+      app_key: "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+      # Lookup the id in the file and ignore errors if the file doesn't exits, so downtime gets created
+      id: "{{ lookup('file', inventory_hostname +'_downtime_id.txt', errors='ignore') }}"
+  - name: Save downtime id to file for later updates and idempotence
+    local_action: copy content="{{ downtime_var.downtime.id }}" dest={{ inventory_hostname +'_downtime_id.txt' }}
```

Comment:
Use `delegate_to` instead of `local_action`. That also improves readability a lot.